### PR TITLE
Tune min_data_in_leaf instead of min_child_samples

### PR DESCRIPTION
**Pull request by Y-oHr-N**
This PR solves the problem mentioned in  https://github.com/optuna/optuna/pull/1018#discussion_r572880664

### CODE REVIEW

**Comment by keisuke-umezawa** [comment]
@toshihikoyanase @crcrpar 
If you have time, could you review it?

File: optuna/integration/_lightgbm_tuner/optimize.py [hunk 1]
```diff
@@ -225,11 +225,11 @@ def _preprocess(self, trial: optuna.trial.Trial) -> None:
...
             self.lgbm_params["bagging_freq"] = trial.suggest_int("bagging_freq", 1, 7)
-        if "min_child_samples" in self.target_param_names:
-            # `GridSampler` is used for sampling min_child_samples value.
+        if "min_data_in_leaf" in self.target_param_names:
+            # `GridSampler` is used for sampling min_data_in_leaf value.
             # The value 1.0 for the hyperparameter is always sampled.
-            param_value = trial.suggest_int("min_child_samples", 5, 100)
-            self.lgbm_params["min_child_samples"] = param_value
+            param_value = trial.suggest_int("min_data_in_leaf", 5, 100)
+            self.lgbm_params["min_data_in_leaf"] = param_value
```
Comment (toshihikoyanase) [inline optuna/integration/_lightgbm_tuner/optimize.py:232]: Could you copy the value of `min_data_in_leaf` value to `min_child_samples` here? If we copy it in the post-processing, the `min_child_samples` value is empty in the best booster.

```suggestion
            self.lgbm_params["min_data_in_leaf"] = param_value

        if "min_data_in_leaf" in self.lgbm_params:
            self.lgbm_params["min_child_samples"] = self.lgbm_params["min_data_in_leaf"]
```

Comment (toshihikoyanase) [reply optuna/integration/_lightgbm_tuner/optimize.py:232]: In addition, please add a test case to ensure that the optimized parameters have `min_child_samples`.

**Review by toshihikoyanase — COMMENTED** [review]
Thank you for your PR. I added comments on the copy of `min_data_in_leaf` value.
Also, could you rebase the PR to resolve CI errors?

@smly If you have any comments as the original author of LightGBM Tuner, please let us know.

**Comment by crcrpar** [comment]
I understood this pull request is a pull request that fixes the wrong priority order of `min_data_in_leaf` and `min_child_samples`: `min_child_samples` > `min_data_in_leaf` in the current LightGBMTuner implementation even though LightGBM puts `min_data_in_leaf` first.

Is this correct?

**Comment by nzw0301** [comment]
@Y-oHr-N any update? :)

**Pull request closed without merging**